internal/agent/device/lifecycle/manager.go

```diff
@@ -92,12 +102,85 @@ func (m *LifecycleManager) Initialize(ctx context.Context, status *v1alpha1.Devi
 }
 
 func (m *LifecycleManager) Sync(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
-	// TODO: implement
+	// this controller currently does not implement a sync operation
 	return nil
 }
 
 func (m *LifecycleManager) AfterUpdate(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
-	// TODO: implement
+	if current.Decommission == nil && desired.Decommission != nil {
+		m.log.Warn("Detected decommissioning request from flightctl service")
+		if err := m.updateWithStartedCondition(ctx); err != nil {
+			return fmt.Errorf("failed to update decommission started status: %w", err)
```

Comment:
If the device can't connect to the management service to update status do you want to decommission anyway?

Reply:
Yeah, I think we need to separate the actual flow (which should complete no matter what) from the status reports. This is why I was keeping that `errored := false` bool so that execution could complete but we would know which status to send back.

Reply:
Maybe I can instead have an array of errors we can return at the end, I have seen that elsewhere in the code.

Reply:
Updated